pkg/util/stringreplace/object.go

```diff
@@ -73,14 +95,27 @@ func visitUnsettableValues(typeOf reflect.Type, original reflect.Value, visitor
 	}
 	switch existing.Kind() {
 	case reflect.String:
-		s := visitor(existing.String())
-		val.Set(reflect.ValueOf(s))
+		s, asString := visitor(existing.String())
+
+		var b []byte
+		if asString {
+			b = []byte(fmt.Sprintf("{\"k\":%q}", s))
+		} else {
+			b = []byte(fmt.Sprintf("{\"k\":%s}", s))
+		}
+		var data map[string]interface{}
```

Comment:
and i don't understand the var data interface{} suggestion.  i definitely don't know the json mechanisms as well as you, but i thought i needed to decode into a map[string]interface in order to get the library to decode the arbitrary json string into a proper go type.  If i just decode a string (not part of a json map), it will always get turned into a string, not a specific data type.

that is, json.Unmarshal("true",&data) is never going to give me the boolean value that i need in some cases (and in other cases i need it to give me the string type)

if i'm still wrong let's chat tomorrow.